File: lib/data/models/chat_message.dart
import 'transaction.dart';

enum MessageRole { user, assistant }

class ChatMessage {
  final String id;
  final MessageRole role;
  final String content;
  final DateTime timestamp;
  final AppTransaction? parsedTransaction;
  final bool isError;

  const ChatMessage({
    required this.id,
    required this.role,
    required this.content,
    required this.timestamp,
    this.parsedTransaction,
    this.isError = false,
  });

  Map<String, dynamic> toMap() => {
    'id': id,
    'role': role.name,
    'content': content,
    'intent': 'conversation',
    'timestamp': timestamp.millisecondsSinceEpoch,
    'parsed_transaction_id': parsedTransaction?.id,
    'is_error': isError ? 1 : 0,
  };

  factory ChatMessage.fromMap(Map<String, dynamic> m) => ChatMessage(
    id: m['id'] as String,
    role: MessageRole.values.firstWhere(
      (e) => e.name == m['role'],
      orElse: () => MessageRole.assistant,
    ),
    content: m['content'] as String,
    timestamp: DateTime.fromMillisecondsSinceEpoch(m['timestamp'] as int),
    isError: (m['is_error'] as int? ?? 0) == 1,
  );
}

File: backend/src/models/ChatMessage.js
const mongoose = require('mongoose');

const chatMessageSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  role: { type: String, required: true, enum: ['user', 'assistant'] },
  content: { type: String, required: true },
  intent: { type: String, default: '' },
  timestamp: { type: Date, default: Date.now },
  parsedTransactionId: { type: String, default: null },
  isError: { type: Boolean, default: false }
}, { _id: false, timestamps: false });

chatMessageSchema.index({ timestamp: -1 });

module.exports = mongoose.model('ChatMessage', chatMessageSchema);
